internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package secrets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	preview_secret_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-06-13/client/secret_service"
+
+	"github.com/hashicorp/hcp/internal/pkg/cmd"
+	"github.com/hashicorp/hcp/internal/pkg/flagvalue"
+	"github.com/hashicorp/hcp/internal/pkg/format"
+	"github.com/hashicorp/hcp/internal/pkg/heredoc"
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+func NewCmdCreate(ctx *cmd.Context, runF func(*CreateOpts) error) *cmd.Command {
+	opts := &CreateOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		IO:            ctx.IO,
+		Output:        ctx.Output,
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "create",
+		ShortHelp: "Create a new static secret.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets secrets create" }} command creates a new static secret under an Vault Secrets App.
+
+		Once the secret is created, it can be read using
+		{{ template "mdCodeOrBold" "hcp vault-secrets secrets read" }} subcommand.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Create new secret in Vault Secrets application on active profile:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets secrets create secret-1 --data-file=/tmp/secrets1.txt
+				`),
+			},
+			{
+				Preamble: `Create secret in different Vault Secrets application, not active profile:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithNoWrap()).Must(`
+				$ hcp vault-secrets secrets create secret-2 --app-name test-app --secret_file=/tmp/secrets2.txt
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "SECRET_NAME",
+					Documentation: "The name of the secret to create.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "data-file",
+					DisplayValue: "DATA_FILE_PATH",
+					Description:  "Absolute path to the secrets data file.",
```

Comment:
Feels like there are three cases:

1. Read from file.
2. Read interactively
3. I would suggest a third case which is reading from STDIN directly: `run some command | hcp vs secret create test --data-file=-` 

It is very common to use `-` to read in from STDIN. I would suggest you follow that pattern.

Reply:
For 3, checking [here](https://hashicorp.slack.com/archives/C0630MA5H8B/p1715632751472559?thread_ts=1715632628.014639&cid=C0630MA5H8B) on slack on how to achieve it

Reply:
added a commit to address [this](https://github.com/hashicorp/hcp/pull/65/commits/32126a5eeac719b49f9c3b58e0f169ab962dbe08) 

```
> make go/build && print "../secret_data_file I ma here" |./bin/hcp vs secrets create test10 --data-file=-
ERROR: X failed to get data file info: stat ../secret_data_file I ma here: no such file or directory

> make go/build && print "../secret_data_file" |./bin/hcp vs secrets create test10 --data-file=- 
App Name  Latest Version  Created At                
test10    3               2024-05-13T20:11:22.205Z  
```

Reply:
addressed in the latest commit.